lib/user_name_suggester.rb

```diff
@@ -21,12 +21,16 @@ def self.find_available_username_based_on(name, allow_username = nil)
     name = fix_username(name)
     i = 1
     attempt = name
-    until attempt == allow_username || User.username_available?(attempt)
+    until attempt == allow_username || User.username_available?(attempt) || i > 100
       suffix = i.to_s
       max_length = User.username_length.end - suffix.length - 1
       attempt = "#{name[0..max_length]}#{suffix}"
       i += 1
     end
+    until attempt == allow_username || User.username_available?(attempt) || i > 200
```

Comment:
Based on the discussion in 

https://github.com/discourse/discourse/pull/4728#issuecomment-282862287. I think this should be `i > 100`?